framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -131,6 +131,12 @@ public static AnnotatedTypeMirror createType(
     /** Actual type wrapped with this AnnotatedTypeMirror. */
     protected final TypeMirror underlyingType;
 
+    /**
+     * Used for generating the hash code for this. Call {@link #getUnderlyingTypeString()} rather
+     * than using the field directly.
+     */
+    private String underlyingTypeString = null;
```

Comment:
It would be more efficient to compute the hash code once and store it.  Storing the string, which forces the client to always recompute the string's hash code.  (You can pass an int to `Objects.hashCode`)
But, if you want it this way anyway, that's OK.